File: templates/types/llamaindexserver/fastapi/generate.py
import logging
import os
import sys

from dotenv import load_dotenv
from llama_index.llms.openai import OpenAI

logging.basicConfig(level=logging.INFO)
logger = logging.getLogger()


def generate_index():
    """
    Index the documents in the data directory.
    """
    from app.index import STORAGE_DIR
    from app.settings import init_settings
    from llama_index.core.indices import (
        VectorStoreIndex,
    )
    from llama_index.core.readers import SimpleDirectoryReader

    load_dotenv()
    init_settings()

    logger.info("Creating new index")
    # load the documents and create the index
    reader = SimpleDirectoryReader(
        os.environ.get("DATA_DIR", "data"),
        recursive=True,
    )
    documents = reader.load_data()
    index = VectorStoreIndex.from_documents(
        documents,
        show_progress=True,
    )
    # store it for later
    index.storage_context.persist(STORAGE_DIR)
    logger.info(f"Finished creating new index. Stored in {STORAGE_DIR}")


def generate_ui_for_workflow():
    """
    Generate UI component for events from workflow.
    Takes --input_file and --output_file as command line arguments.
    """
    import asyncio

    from llama_index.server.gen_ui.main import generate_ui_for_workflow

    # Parse command line arguments
    args = sys.argv[1:]
    input_file = None
    output_file = None

    for i in range(0, len(args), 2):
        if args[i] == "--input_file":
            input_file = args[i + 1]
        elif args[i] == "--output_file":
            output_file = args[i + 1]

    if not input_file or not output_file:
        print("Error: Both --input_file and --output_file arguments are required")
        sys.exit(1)

    llm = OpenAI(model="gpt-4.1")
    code = asyncio.run(generate_ui_for_workflow(input_file, llm=llm))
    with open(output_file, "w") as f:
        f.write(code)

File: packages/create-llama/templates/components/use-cases/typescript/financial_report/src/generate.ts
import { SimpleDirectoryReader } from "@llamaindex/readers/directory";
import "dotenv/config";
import { storageContextFromDefaults, VectorStoreIndex } from "llamaindex";
import { initSettings } from "./app/settings";

async function generateDatasource() {
  console.log(`Generating storage context...`);
  // Split documents, create embeddings and store them in the storage context
  const storageContext = await storageContextFromDefaults({
    persistDir: "storage",
  });
  // load documents from current directory into an index
  const reader = new SimpleDirectoryReader();
  const documents = await reader.loadData("data");

  await VectorStoreIndex.fromDocuments(documents, {
    storageContext,
  });
  console.log("Storage context successfully generated.");
}

(async () => {
  const args = process.argv.slice(2);
  const command = args[0];

  initSettings();

  if (command === "ui") {
    console.error("This project doesn't use any custom UI.");
    return;
  } else {
    if (command !== "datasource") {
      console.error(
        `Unrecognized command: ${command}. Generating datasource by default.`,
      );
    }
    await generateDatasource();
  }
})();
